### PR TITLE
Add keccak256 and keccak512 native support

### PR DESCRIPTION
**Pull request by k06a**
Related to: https://github.com/nearprotocol/nearcore/pull/2045

### CODE REVIEW

**Comment by vgrichina** [comment]
@k06a Please make sure there is corresponding PR to https://github.com/nearprotocol/near-runtime-ts as well

**Comment by k06a** [comment]
@vgrichina thanks, https://github.com/nearprotocol/near-runtime-ts/pull/120

File: near-bindgen/Cargo.toml [hunk 1]
```diff
@@ -21,8 +21,8 @@ path = "compilation_tests/all.rs"
 serde = { version = "1.0", features = ["derive"] }
 near-bindgen-macros = { path = "../near-bindgen-macros", version = "0.3.10"}
 borsh = "0.2.10"
-near-vm-logic = "0.4.3"
-near-runtime-fees = "0.4.3"
+near-vm-logic = { path = "../../nearcore/runtime/near-vm-logic" }
```
[the remark this answers is not in the document]
Comment (k06a) [reply near-bindgen/Cargo.toml]: Gonna revert these changes, ofcourse

File: rust-toolchain [hunk 2]
```diff
@@ -1 +1 @@
-stable
+nightly
```
[the remark this answers is not in the document]
Comment (k06a) [reply rust-toolchain]: Had some incompatibilities during referencing by `path`

**Comment by k06a** [comment]
@evgenykuzyakov @nearmax  since this feature requires changes in multiple repos, IDK how to specify require versions to keep everything compilable. Should I reference by git and branch?

File: rust-toolchain (hunk 2 above)
[the remark this answers is not in the document]
Comment (nearmax) [reply rust-toolchain]: I suspect this is a wrong place. I don't see why `near-bindgen` should be compiled with nightly. I suspect that some libraries that you are using in near-bridge might require nightly though. We should be change rust toolchain in near-bridge repo not here.

**Pull request merged**